modules/tactical/src/mutation.ts

```diff
@@ -0,0 +1,119 @@
+/**
+ * Represents a mutation to be performed on top of an object.
+ */
+export interface Mutation {
+  
+  /**
+   * Applies the mutation to an object, returning success or failure.
+   */
+  apply(obj: any): boolean;
+}
+
+/**
+ * Sets a group of properties on an object, overwriting any previous values.
+ */
+export class SetPropertiesMutation implements Mutation {
+  changes: any;
+  
+  constructor(changes: any) {
+    this.changes = changes;
+  }
```

Comment:
I think the idiomatic `constructor(public changes: any) {}` (which would replace lines 16 - 20), is preferred in angular/angular.